ctru-rs/src/mii.rs

```diff
@@ -0,0 +1,479 @@
+    pub color: u8,
+    pub scale: u8,
+    pub y_scale: u8,
+    pub rotation: u8,
+    pub x_spacing: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct EyebrowDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_scale: u8,
+    pub rotation: u8,
+    pub x_spacing: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct NoseDetails {
+    pub style: u8,
+    pub scale: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MouthDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_scale: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MustacheDetails {
+    pub mouth_y_position: u8,
+    pub mustache_style: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct BeardDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct GlassDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MoleDetails {
+    pub is_enabled: bool,
+    pub scale: u8,
+    pub x_position: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MiiData {
+    pub mii_options: MiiDataOptions,
+    pub mii_selector_position: SelectorPosition,
+    pub mii_console_identity: ConsoleIdentity,
+
+    pub system_id: [u8; 8],
+    pub mac_address: [u8; 6],
+
+    pub mii_details: Details,
+    pub mii_name: String,
+
+    pub height: u8,
+    pub width: u8,
+
+    pub face_style: FaceStyle,
+    pub face_details: FaceDetails,
+
+    pub hair_style: u8,
+
+    pub hair_details: HairDetails,
+    pub eye_details: EyeDetails,
+    pub eyebrow_details: EyebrowDetails,
+    pub nose_details: NoseDetails,
+    pub mouth_details: MouthDetails,
+    pub mustache_details: MustacheDetails,
+    pub beard_details: BeardDetails,
+    pub glass_details: GlassDetails,
+    pub mole_details: MoleDetails,
+
+    pub author_name: String,
+}
+
+impl From<[u8; 92]> for MiiData {
+    fn from(raw_mii_data: [u8; 92]) -> Self {
+        // Source for the representation and what each thing means: https://www.3dbrew.org/wiki/Mii
+        let raw_options = vec_bit(raw_mii_data[0x1]);
+        let raw_position = vec_bit(raw_mii_data[0x2]);
+        let raw_device = vec_bit(raw_mii_data[0x3]);
+        let system_id = [
+            raw_mii_data[0x4],
+            raw_mii_data[0x5],
+            raw_mii_data[0x6],
+            raw_mii_data[0x7],
+            raw_mii_data[0x8],
+            raw_mii_data[0x9],
+            raw_mii_data[0xA],
+            raw_mii_data[0xB],
+        ];
+        let creator_mac = [
+            raw_mii_data[0x10],
+            raw_mii_data[0x11],
+            raw_mii_data[0x12],
+            raw_mii_data[0x13],
+            raw_mii_data[0x14],
+            raw_mii_data[0x15],
+        ];
+        let raw_details: [bool; 16] =
+            get_and_concat_vec_bit(raw_mii_data.as_slice(), &[0x18, 0x19])
```

Comment:
nit: it's more idiomatic to do `&raw_mii_data` instead of `raw_mii_data.as_slice()`